### DISCUSSION TITLE
[Proposal] C# GUID literal (companion to VB proposal)

### DISCUSSION THREAD

**Discussion by mkane91301** [dotnet/csharplang#985] [General]
[vblang #27](https://github.com/dotnet/vblang/issues/27) proposes a syntax for GUID literals and suggests CLR changes to enable GUID constants. This proposal is to match the feature for C# with appropriate syntax.

**Comment by dsaf** [comment]
We don't even have DateTime literals.

Also RIP https://github.com/dotnet/roslyn/issues/263

**Comment by tannergooding** [comment]
I would much rather see support for blittable constants first so that literals are actually efficient. Outside of that, calling the constructor that takes a string (or using a code refactoring to convert that to the raw value overload) seems to work well enough.
